packages/react/src/useSecurityKeys.ts

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from 'react'
+
+import {
+  ActionErrorState,
+  ActionSuccessState,
+  AddSecurityKeyHandlerResult,
+  addSecurityKeyPromise,
+  ErrorPayload,
+  OTHER_ERROR_CODE
+} from '@nhost/core'
+
+import { useNhostClient } from './useNhostClient'
+
+interface AddSecurityKeyHandler {
+  (
+    /** Optional human-readable name of the security key */
+    nickname?: string
+  ): Promise<AddSecurityKeyHandlerResult>
+}
+interface RemoveSecurityKeyHandlerResult extends ActionErrorState, ActionSuccessState {}
+interface RemoveSecurityKeyHandler {
+  (
+    /** Unique identifier of the security to remove */
+    id: string
+  ): Promise<RemoveSecurityKeyHandlerResult>
+}
+
+export type SecurityKey = { id: string; nickname?: string }
+
+export interface SecurityKeysHookResult extends ActionErrorState, ActionSuccessState {
+  /** Add a security key to the current user with the WebAuthn API */
+  add: AddSecurityKeyHandler
+  /**  List the security keys of the current user */
+  list: SecurityKey[]
+  /** Remove the given security key from the list of allowed keys for the current user */
+  remove: (id: string) => Promise<RemoveSecurityKeyHandlerResult>
+}
+
+interface SecurityKeysHook {
+  (): SecurityKeysHookResult
+}
+
+/**
+ * Use the hook `useSecurityKeys` to list, add and remove the WebAuthn security keys of the user.
+ *
+ * When WebAuthn is enabled, the `add` function will work as expected.
+ *
+ * You have to make sure the current user has the right Hasura permissions on the `auth.user_authenticators` table to list and/or remove their keys:
+ *
+ * - List keys: permissions to select the `id` and `nickname` columns
+ * - Remove keys: permissions to delete rows
+ *
+ * It is recommended add the custom chck `{ user_id: { _eq: "X-Hasura-User-Id" } }` to make sure the user can only list and remove their own keys.
+ *
+ * @example
+ * ```tsx
+ * const { add, list, remove, isLoading, isSuccess, isError, error } = useSecurityKeys()
+ *
+ * return (
+ *   <ul>
+ *     {list.map(({ id, nickname }) => (
+ *       <li key={id} onClick={() => remove(id)}>{nickname}</li>
+ *     ))}
+ *   </ul>
+ * )
+ * ```
+ *
+ * @docs https://docs.nhost.io/reference/react/use-security-keys
+ */
+export const useSecurityKeys: SecurityKeysHook = () => {
+  const nhost = useNhostClient()
+  const [list, setList] = useState<SecurityKey[]>([])
+  const [error, setError] = useState<ErrorPayload | null>(null)
+  const isSuccess = !error
+  const isError = !!error
+
+  const [isLoading, setIsLoading] = useState(false)
+
+  useEffect(() => {
+    const loadList = async () => {
+      setIsLoading(true)
+      // * Initial list
+      const userId = nhost?.auth.getUser()?.id
+      if (userId) {
+        const query = `
```

Comment:
I have a feeling that we should use `graphql-tag` for defining GraphQL queries / mutations.

Reply:
for development purpose, yes. but I didn't here for two reasons
- graphql is not a dependency
- with a Graphql tag, it would 1. parse the string 2. stringify the document node to end up with the same result -> unnecessary computation

Reply:
Yea, I think it's fine to skip `grpahql-tag`.

What we could start doing is add `#graphql` to the query string as many IDEs, then highlights the code as GraphQL:

<img width="389" alt="CleanShot 2022-09-06 at 21 02 24@2x" src="https://user-images.githubusercontent.com/331818/188717578-81d58b3e-96ee-4141-a907-8435d7d064f1.png">